ffi/examples/read-table/schema.h

```diff
@@ -195,30 +213,33 @@ void free_builder(SchemaBuilder builder) {
 }
 
 // Print the schema of the snapshot
-void print_schema(SharedSnapshot* snapshot) {
+void print_schema(SharedSnapshot* snapshot)
+{
   print_diag("Building schema\n");
   SchemaBuilder builder = {
     .list_count = 0,
     .lists = NULL,
   };
-  EngineSchemaVisitor visitor = { .data = &builder,
-                                  .make_field_list = make_field_list,
-                                  .visit_struct = visit_struct,
-                                  .visit_array = visit_array,
-                                  .visit_map = visit_map,
-                                  .visit_decimal = visit_decimal,
-                                  .visit_string = visit_string,
-                                  .visit_long = visit_long,
-                                  .visit_integer = visit_integer,
-                                  .visit_short = visit_short,
-                                  .visit_byte = visit_byte,
-                                  .visit_float = visit_float,
-                                  .visit_double = visit_double,
-                                  .visit_boolean = visit_boolean,
-                                  .visit_binary = visit_binary,
-                                  .visit_date = visit_date,
-                                  .visit_timestamp = visit_timestamp,
-                                  .visit_timestamp_ntz = visit_timestamp_ntz };
+  EngineSchemaVisitor visitor = {
```

Comment:
aside: We had discussed in the past how adding new non-nullable struct members could silently break things; and even doing a runtime check would be annoying because the problem wouldn't manifest until the code ran.

One idomatic alternative might be an initializer function that takes all the arguments and stuffs them into a struct instance that gets returned. The arguments are non-defaulted and so any change in arg count would produce compilation errors as desired. We can also stop people from initializing the struct directly if we make it opaque+malloc+free, but I'm not sure that's worth the trouble. Seems like giving people a way to do it safely is enough for those who care about safety?

Reply:
Yeah, that's a nice idea. Created #268